src/py21cmfast/wrapper/classy_interface.py

```diff
@@ -210,7 +285,145 @@ def compute_rms(
         W_k[kr < 1.0e-3] = 1.0 - 3.0 * (kr_small**2) / 10.0
 
         integrand = priomordial_PS * (transfer * W_k) ** 2
-        var = intg.simpson(integrand, x=np.log(k_output / k_output.unit))
+        var = intg.simpson(integrand, x=np.log(k_transfer / k_transfer.unit))
         rms_list.append(np.sqrt(var))
     # NOTE: intg.simpson removes the unit information, which is why we multiply by the unit when we return
     return np.array(rms_list) * transfer.unit
+
+
+def find_redshift_kinematic_decoupling(classy_output: Class) -> float:
+    """
+    Find the redshift of kinematic decoupling.
+
+    For simplicity, we approximate the redshift of kinematic decoupling to be the same redshift of recombination,
+    which is defined as the moment when x_e = n_e/(n_H + n_He) = 0.1. For LCDM with Planck 2018 parameters, this corresponds
+    to z_dec ~ 1070.
+
+    Parameters
+    ----------
+    classy_output : :class:`classy.Class`
+        An object containing all the information from the CLASS calculation.
+
+    Returns
+    -------
+    z_dec : float
+        Redshift of kinematic decoupling.
+    """
+    YHe = classy_output.get_current_derived_parameters(["YHe"])["YHe"]
+    z_array = np.linspace(800, 1200, 400)
+    # There is a need to multiply by n_H/(n_H+n_He)=(1-YHe)/(1-(1-1/_not4_)*YHe)
+    # because CLASS returns n_e/n_H (but we want n_e/(n_H+n_He))
+    x_e_array = (
+        np.array([classy_output.ionization_fraction(z) for z in z_array])
+        * (1.0 - YHe)
+        / (1.0 - (1.0 - 1.0 / _not4_) * YHe)
+    )
+    z_dec = interp1d(x_e_array, z_array, kind="cubic")(0.1)
+    return z_dec
+
+
+class EHTransferFunction:
+    """Eisenstein & Hu (1998) transfer function matching 21cmFAST implementation. This is the MDM (massive neutrino) formulation from their Eq. 29-31."""
+
+    def __init__(self, classy_output):
+        """
+        Initialize EH transfer function parameters.
+
+        Parameters
+        ----------
+        classy_output : :class:`classy.Class`
+            An object containing all the information from the CLASS calculation.
+        """
+        self.Omega_m = classy_output.Omega_m()
+        self.Omega_b = classy_output.Omega_b()
+        self.hlittle = classy_output.h()
+        self.T_cmb = classy_output.T_cmb
+        self.Omega_nu = classy_output.Omega_nu()
+        self.N_nu = classy_output.pars["N_ncdm"]
+
+        self.omhh = self.Omega_m * self.hlittle**2
+        self.obhh = self.Omega_b * self.hlittle**2
+        self.theta_cmb = self.T_cmb / 2.7
+
+        self.f_nu = max(self.Omega_nu / self.Omega_m, 1e-10)
+        self.f_baryon = max(self.Omega_b / self.Omega_m, 1e-10)
+
+        self._set_parameters()
+
+    def _set_parameters(self):
+        """Set constants used in the EH transfer function (matches TFset_parameters in cosmology.c)."""
+        f_nu = self.f_nu
```

Comment:
I wonder actually if it's not just better to depend on `hmf`... it already has the EH transfer function in it, and also a CAMB function, and BBKS, and...